ghmirror/data_structures/redis_data_structures.py

```diff
@@ -0,0 +1,80 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: Red Hat Inc. 2020
+# Author: Maha Ashour <mashour@redhat.com>
+
+"""
+Caching data in Redis.
+"""
+
+import os
+import pickle
+import redis
+
+
+class RedisCache:
+    """
+    Dictionary-like implementation for caching requests in Redis.
+    """
+    def __init__(self):
+        endpoint = os.environ.get('REDIS_ENDPOINT')
+        port = os.environ.get('REDIS_PORT', 6379)
+        self.cache = redis.Redis(
+            host=endpoint, port=port)
+
+    def __contains__(self, item):
+        sr_key = serialize(item)
+        return self.cache.exists(sr_key)
+
+    def __getitem__(self, item):
+        sr_key = serialize(item)
+        sr_value = self.cache.get(sr_key)
+        if sr_value is None:
+            raise KeyError(item)
+        return deserialize(sr_value)
+
+    def __setitem__(self, key, value):
+        sr_key = serialize(key)
+        sr_value = serialize(value)
+        self.cache.set(sr_key, sr_value)
+
+    def __iter__(self):
+        return self.scan_iter()
+
+    def scan_iter(self):
```

Comment:
is this really meant to be a public method?

Reply:
no, I didn't quite realize methods could be public or private in Python, but I understand now that the __ prefix makes it private. thanks for pointing this out.